npm/design-system/src/components/SpecList/SpecList.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+import {
+  FileComponentProps,
+  FolderComponentProps,
+  FileExplorer,
+  FileExplorerProps,
+} from '../FileExplorer/FileExplorer'
+import { makeFileHierarchy } from '../FileExplorer/helpers/makeFileHierarchy'
+
+import { InlineIcon } from '@iconify/react'
+import javascriptIcon from '@iconify/icons-vscode-icons/file-type-js-official'
+import typescriptIcon from '@iconify/icons-vscode-icons/file-type-typescript-official'
+import reactJs from '@iconify/icons-vscode-icons/file-type-reactjs'
+import reactTs from '@iconify/icons-vscode-icons/file-type-reactts'
+import folderClosed from '@iconify/icons-vscode-icons/default-folder'
+import folderOpen from '@iconify/icons-vscode-icons/default-folder-opened'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import styles from './SpecList.module.scss'
+
+library.add(fas)
+
+const icons: Record<string, any> = {
```

Comment:
This doesn't need a declared type. It will be correctly inferred (though if you want the exact types of the icons, you need to follow each icon object with `as const`)

Reply:
I tried this (in combination with typing `getExt` better, see below comment) but cannot get something without type casts. Can you show me how to do this?